### PR TITLE
feat: Allow skipping of validation when combining workspaces

### PR DESCRIPTION
**Pull request by lorenzennio**
# Description

It would be nice to include the option to avoid the validation step when combining two workspaces:
https://github.com/scikit-hep/pyhf/blob/a1a31f17967740c93244f5d9ad17aea1fbef6be3/src/pyhf/workspace.py#L709

# Checklist Before Requesting Reviewer

- [x] Tests are passing
- [x] "WIP" removed from the title of the pull request
- [x] Selected an Assignee for the PR to be responsible for the log summary

# Before Merging

For the PR Assignees:

- [x] Summarize commit messages into a comprehensive review of the PR

```
* Add validate arg to workspace.combine to allow for optionally skipping
  workspace validation during. Default is to validate.
* Add test for validate arg being used.
* Add Lorenz Gaertner to contributors list.
```

### CODE REVIEW

**Comment by kratsg** [comment]
Is there a reason you want to combine workspaces that don't produce a valid workspace?

**Comment by lorenzennio** [comment]
Yes, I am working with workspaces that contain instructions for a custom modifier. These don't pass the validation in the first place, but for single workspaces I can simply skip it. Now I want to combine two of them, for a global fit.

**Review by kratsg — APPROVED** [review]

File: src/pyhf/workspace.py [hunk 1]
```diff
@@ -770,7 +773,7 @@ def combine(cls, left, right, join='none', merge_channels=False):
             'observations': new_observations,
             'version': new_version,
         }
-        return cls(newspec)
+        return cls(newspec, validate=validate)
```
Comment (matthewfeickert) [inline src/pyhf/workspace.py:776]: This should get covered by a test. I can add that though unless someone beats me to it.

**Review by matthewfeickert — CHANGES_REQUESTED** [review]

**Review by matthewfeickert — CHANGES_REQUESTED** [review]
@lorenzennio Can you also [add your name](https://github.com/scikit-hep/pyhf/blob/a1a31f17967740c93244f5d9ad17aea1fbef6be3/CONTRIBUTING.md#pull-request-procedure) to `docs
/contributors.rst` below 

https://github.com/scikit-hep/pyhf/blob/a1a31f17967740c93244f5d9ad17aea1fbef6be3/docs/contributors.rst?plain=1#L35

?

**Pull request merged**